test/extensions/filters/http/csrf/csrf_filter_integration_test.cc

```diff
@@ -48,26 +48,26 @@ name: envoy.csrf
 
 class CsrfFilterIntegrationTest : public HttpProtocolIntegrationTest {
 protected:
-  absl::string_view testNormalRequest(Http::TestHeaderMapImpl&& request_headers) {
+  const char* testNormalRequest(Http::TestHeaderMapImpl&& request_headers) {
     initialize();
     codec_client_ = makeHttpConnection(lookupPort("http"));
     auto response = codec_client_->makeRequestWithBody(request_headers, 1024);
     waitForNextUpstreamRequest();
     upstream_request_->encodeHeaders(Http::TestHeaderMapImpl{{":status", "200"}}, true);
     response->waitForEndStream();
 
-    return response->complete() ? response->headers().Status()->value().getStringView()
-                                : absl::string_view("incomplete");
+    return response->complete() ? response->headers().Status()->value().getStringView().data()
```

Comment:
.data() is not nul-terminated. Let's keep these all in string_view form.

You should never access string or string_view .data without also using .size().